### ISSUE TITLE
pip check failes due to inconsistencies in package and recipe dependencies

### ISSUE THREAD

**Issue by jhkennedy** [conda-forge/pydap-feedstock#28]
### Solution to issue cannot be found in the documentation.

- [x] I checked the documentation.

### Issue

`pip check` currently fails for any recipe that depends on `pydap` but doesn't include `beatifulsoup4` or `lxml` as a dependency (directly or indirectly). 

This is because the `pydap` recipe does not include `beautifulsoup4` or `lxml`:
https://github.com/conda-forge/pydap-feedstock/blob/main/recipe/meta.yaml#L32-L39

but they are listed as a requirement in pydap's `pyproject.toml`:
https://github.com/pydap/pydap/blob/3.5.2/pyproject.toml#L27-L28

This can be resolved by depending on `pydap-server` instead, but it'd be good to ensure that the `pyproject.toml` and the package recipes are consistent. Ideally:
*  the `pydap` recipe run requirements would match the `pyproject.toml` dependencies (`python -m pip install pydap`)
* the `pydap-server` recipe would match the `pyproject.toml` optional dependents for `server` (`python -m pip install pydap[server]`)

---

Note: the opposite is true for `docopt-ng` and `jinja`, which are listed as `[server]` optional dependencies in the `pyproject.toml` but are required for `pydap` recipe. 

And `webob` is a requirement in both the recipe and `pyproject.toml` but the comment in the recipe implies it's optional. 

### Installed packages

```shell
You can see this in the raider-feedstock builds here:
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1120416&view=logs&j=d818e662-7181-5ce9-b265-50d03742b7af&t=367317f3-9b77-59ac-9e12-e56370c98acd
```

### Environment info

```shell
You can see this in the raider-feedstock builds here:
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1120416&view=logs&j=d818e662-7181-5ce9-b265-50d03742b7af&t=367317f3-9b77-59ac-9e12-e56370c98acd
```

**Comment by ocefpaf** [comment]
You should be be using pydap-server. See https://github.com/conda-forge/pydap-feedstock/blob/9330c817a65084a3884ac576b4e0873423f3c9fe/recipe/meta.yaml#L51

**Comment by jhkennedy** [comment]
>> This can be resolved by depending on pydap-server instead

> You should be be using pydap-server

@ocefpaf Sorry, maybe I wasn't clear.  `pip check` will always fail for `pydap` right now, unless `lxml` and `beautifulsoup4` happen to be installed, which you can reproduce like:
```shell
$ mamba create -n tmp python pip pydap
$ mamba activate tmp
$ pip check
pydap 3.5.3 requires beautifulsoup4, which is not installed.
pydap 3.5.3 requires lxml, which is not installed.
```

and, again, is caused by a mismatch in the Python package requirements (`pyproject.toml`) and the conda package requirements (`recipe/meta.yaml`) as described above. 

---

Note: I don't fully understand why this issue isn't being picked up in testing since you do have `pip check` implemented for `pydap`:
https://github.com/conda-forge/pydap-feedstock/blob/main/recipe/meta.yaml#L49

But looking @ the build logs for the recent v3.5.3 recipe update, it seems like the tests for `pydap` weren't run:
```shell
TEST START: /home/conda/feedstock_root/build_artifacts/noarch/pydap-3.5.3-pyhd8ed1ab_0.conda
Adding in variants from /tmp/tmpfmeczto6/info/recipe/conda_build_config.yaml
Adding in variants from config.variant
Nothing to test for: /home/conda/feedstock_root/build_artifacts/noarch/pydap-3.5.3-pyhd8ed1ab_0.conda
```
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1136764&view=logs&j=7b6f2c87-f3a7-5133-8d84-7c03a75d9dfc&t=9eb77fd2-8ddd-5444-8fc0-71cb28dcb736&l=1293

**Comment by Mikejmnez** [comment]
Thanks @jhkennedy for raising this issue. Indeed there seems to be a mismatch between the pyproject.toml and the recipe.yaml for pydap's dependencies. I do not understand why this was not picked up during testing in conda - but I haven't looked to deeply and am not very knowledgeable on conda-feedstock stuff. I can regenerate the recipe/meta.yaml for pydap and see if that triggers the testing (at least locally on my machine).

**Comment by jhkennedy** [comment]
@ocefpaf Yes, I am aware you split the package, and I know exactly how to resolve the issue for my recipes and dependent packages, as I've previously stated above, twice. 

**This issue is not about that.** 

Can you please focus on the issue at hand, which is the fact that your recipes do not match the equivalent package targets and that the `pydap` output is *broken.*

That is, from a close re-reading of #23, the goal of splitting the package up was so that:
* `pydap` conda output facilitates pydap *client* usage and is equivalent to `python -m pip install pydap`
* `pydap-server` conda output facilitates pydap *server* usage and is equivalent to `python -m pip install pydap[server,netcdf]` 

and so the dependencies should be consistent:
* `python -m pip install pydap` and `conda install pydap` to provide the same dependencies
* `python -m pip install pydap[server]` and `conda install pydap-server` to provide the same dependencies

Right now, they are not.

Notably, I missed that the `netcdf` target is required as well for `pydap-server` based on this comment:
https://github.com/conda-forge/pydap-feedstock/issues/23#issuecomment-2453650536

So that should *also* be added to the pydap-server output. 

Digging back through the recipe blame, I see that in #22, you removed `beautifulsoup4` and `lxml` from the package recipe, commenting that they are option server dependencies, which they are *not* as defined by the package.

So, were those dependencies causing problems for xarray, as xarray seems to be were the motivation for this came from?

**Comment by ocefpaf** [comment]
> Can you please focus on the issue at hand, which is the fact that your recipes do not match the equivalent package targets and that the `pydap` output is _broken._

I won't respond anymore due to your aggressive tone. All you need to know is in there. Also, conda is not pip and the splitting was requested by the many users of this package. Please be mindful of the volunteers that provide this package, and many more, for free on their own time. Bye for now and good luck.

**Comment by jhkennedy** [comment]
Message received -- I'm sorry my frustration at being ignored repeatedly boiled over into an "aggressive tone," and I agree there are much better uses of *both* our volunteer time. Closing as this is long past productive.

**Comment by ocefpaf** [comment]
You were not ignored. You just could not understand that the model explained to you for this conda package doesn't fit your expectations.
